pkg/services/ngalert/schedule/schedule_test.go

```diff
@@ -395,6 +395,19 @@ func newFakeExternalAlertmanager(t *testing.T) *fakeExternalAlertmanager {
 	return am
 }
 
+func (am *fakeExternalAlertmanager) AlertNamesCompare(expected []string) bool {
+	n := []string{}
+	for _, a := range am.Alerts() {
+		for k, v := range a.Alert.Labels {
+			if k == model.AlertNameLabel {
+				n = append(n, v)
```

Comment:
Does n need to built and `cmp` compared? Would if `alerts := am.Alerts()`, then returning false if they are not the same length, and return false as soon as we see one that doesn't match work?

Reply:
I've added an early return - re the second part, I thought about turning `expected` into a set for faster lookups but seemed an overkill for a function that's only used once and within tests.

Reply:
Oh I didn't realize I was in a test file :-/ Sorry about that.